source/lac/petsc_parallel_block_sparse_matrix.cc

```diff
@@ -18,6 +18,38 @@
 
 #ifdef DEAL_II_WITH_PETSC
 
+// A dummy utility routine to create an empty matrix in case we import
+// a MATNEST with NULL blocks
+static void
+createDummyMat(MPI_Comm comm,
+               PetscInt lr,
+               PetscInt gr,
+               PetscInt lc,
+               PetscInt gc,
+               Mat *    dummy)
```

Comment:
The C++ style is to put functions into an anonymous namespace instead of making functions `static`. Let's also use use deal.II style for names:
```suggestion
namespace {
// A dummy utility routine to create an empty matrix in case we import
// a MATNEST with NULL blocks
static void
create_dummy_matrix(MPI_Comm comm,
               PetscInt local_rows,
               PetscInt global_rows,
               PetscInt local_columns,
               PetscInt global_columns,
               Mat *    dummy)
```
Extra points if you let the function return the `Mat` object by value, rather than via the last argument :-)

Reply:
done